dbms/src/Common/ThreadStatus.cpp

```diff
@@ -50,6 +50,19 @@ ThreadStatus::ThreadStatus()
 
 ThreadStatus::~ThreadStatus()
 {
+    try
+    {
+        if (untracked_memory > 0)
+            memory_tracker.alloc(untracked_memory);
+        else
+            memory_tracker.free(-untracked_memory);
+    }
+    catch (const DB::Exception &)
+    {
+        /// It's a minor tracked memory leak here (not the memory itself but it's counter).
```

Comment:
It looks strange that the counter update can throw. I'd move fault injection from MemoryTracker::alloc() to a separate function.

Reply:
MemoryTracker throws by design if it cannot alloc some memory. There's an optimisation upper that allows to alloc some more memory that we cannot track. It's not interesting for our thread memory tracker here cause it is dying but it's probably interesting for its parent.